airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java

```diff
@@ -168,10 +173,21 @@ protected AutoCloseableIterator<JsonNode> queryTableFullRefresh(final BigQueryDa
                                                                   final SyncMode syncMode,
                                                                   final Optional<String> cursorField) {
     LOGGER.info("Queueing query for table: {}", tableName);
-    return queryTable(database, String.format("SELECT %s FROM %s",
-        enquoteIdentifierList(columnNames, getQuoteString()),
-        getFullyQualifiedTableNameWithQuoting(schemaName, tableName, getQuoteString())),
-        tableName, schemaName);
+     // This corresponds to the initial sync for in INCREMENTAL_MODE, where the ordering of the records matters
+    // as intermediate state messages are emitted (if the connector emits intermediate state).
+    if (syncMode.equals(SyncMode.INCREMENTAL) && getStateEmissionFrequency() > 0) {
+      final String quotedCursorField = enquoteIdentifier(cursorField.get(), getQuoteString());
+      return queryTable(database, String.format("SELECT %s FROM %s ORDER BY %s ASC",
```

Comment:
if its incremental where is > lastState?

Reply:
No it is not incremental it is just the first sync( full sync)  when we sync a stream incrementaly which not have a state.